staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -1137,21 +1173,42 @@ func newHTTP2ServerWithClient(handler http.Handler, clientTimeout time.Duration)
 	}
 }
 
+type headerMatcher struct {
+	lock    sync.Mutex
+	errsGot []error
+}
+
 // verifies that the expected flow schema and priority level UIDs are attached to the header.
-func expectMatchingAPFHeaders(t *testing.T, w http.ResponseWriter, expectedFS, expectedPL string) {
-	if w == nil {
-		t.Fatal("expected a non nil HTTP response")
```

Comment:
The new behavior no longer calls Fatal at all on these errors? If that is intended, you could have just changed these to e.g. call t.Error() and return.

Reply:
yes, i don't think we need to call `Fatal` for these errors, and I also wanted all the test asserts to be co-located at the end of the test, so it's easier to follow. (but I don't have any strong opinion here)